components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java

```diff
@@ -262,4 +267,91 @@ private String logMessageDetails(MessageContext messageContext, long elapsedTime
         return logMessage;
     }
 
+    private void setAPIParametersToMessageContext(MessageContext messageContext) {
+
+        AuthenticationContext authContext = APISecurityUtils.getAuthenticationContext(messageContext);
+        org.apache.axis2.context.MessageContext axis2MsgContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+
+        String consumerKey = "";
+        String username = "";
+        String applicationName = "";
+        String applicationId = "";
+        if (authContext != null) {
+            consumerKey = authContext.getConsumerKey();
+            username = authContext.getUsername();
+            applicationName = authContext.getApplicationName();
+            applicationId = authContext.getApplicationId();
+        }
+
+        String context = (String) messageContext.getProperty(RESTConstants.REST_API_CONTEXT);
+        String api_version = (String) messageContext.getProperty(RESTConstants.SYNAPSE_REST_API);
```

Comment:
Fixed in #1494 (https://github.com/wso2/carbon-apimgt/commit/68b378607966da79d994692e05c46be213f2ecc0)